airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3ConsumerFactory.kt

```diff
@@ -150,6 +162,39 @@ class S3ConsumerFactory {
                 }
                 LOGGER.info { "Cleaning up destination completed." }
             }
+
+            // On stream success clean up the objects marked for deletion per stream. This is done
+            // on per-stream basis
+            streamSyncSummaries.forEach { entry: Map.Entry<StreamDescriptor, StreamSyncSummary> ->
+                val streamSuccessful =
+                    entry.value.terminalStatus ==
+                        AirbyteStreamStatusTraceMessage.AirbyteStreamStatus.COMPLETE
+                if (streamSuccessful) {
+                    val writeConfig = streamDescriptorToWriteConfig[entry.key]
+                    writeConfig?.let {
```

Comment:
nit: `!!`? Presumably, if the writeconfig is null, then something is very wrong, and we should fail loudly instead of silently nooping

Reply:
done